internal/iapl/policy.go

```diff
@@ -127,6 +140,47 @@ func NewPolicyFromFile(filePath string) (Policy, error) {
 	return NewPolicy(policy), nil
 }
 
+// NewPolicyFromFiles reads the provided file paths, merges them, and returns a new Policy.
+func NewPolicyFromFiles(filePaths []string) (Policy, error) {
+	mergedPolicy := PolicyDocument{}
+
+	for _, filePath := range filePaths {
+		file, err := os.Open(filePath)
+		if err != nil {
+			return nil, err
+		}
+		defer file.Close()
+
+		var filePolicy PolicyDocument
+
+		if err := yaml.NewDecoder(file).Decode(&filePolicy); err != nil {
+			return nil, err
+		}
+
+		mergedPolicy = mergedPolicy.MergeWithPolicyDocument(filePolicy)
+	}
+
+	return NewPolicy(mergedPolicy), nil
+}
+
+// NewPolicyFromDirectory reads the provided directory path, reads all files in the directory, merges them, and returns a new Policy.
+func NewPolicyFromDirectory(directoryPath string) (Policy, error) {
+	files, err := os.ReadDir(directoryPath)
+	if err != nil {
+		return nil, err
+	}
+
+	filePaths := make([]string, 0, len(files))
+
+	for _, file := range files {
+		if !file.IsDir() {
+			filePaths = append(filePaths, directoryPath+"/"+file.Name())
+		}
+	}
```

Comment:
Since we only accept yaml files, we might want to exclude non `.yml` or `.yaml` extensions.